frontend_app_gallery/src/container/Gallery.jsx

```diff
@@ -164,34 +166,43 @@ class Gallery extends React.Component {
     // app crash telling it cannot render a Link outside a router
     // see https://github.com/tracim/tracim/issues/1637
     // GLOBAL_dispatchEvent({type: 'setBreadcrumbs', data: {breadcrumbs: breadcrumbsList}})
-    return breadcrumbsList
+    this.setState({ breadcrumbsList, breadcrumbsLoaded: !includeFile })
   }
 
-  loadFolderDetail = async (workspaceId, folderId) => {
+  loadFolderDetailAndParentsDetails = async (workspaceId, folderId) => {
     const { state, props } = this
 
+    const folderDetail = {}
+
     let fetchContentDetail = await handleFetchResult(
       await getFolderDetail(state.config.apiUrl, workspaceId, folderId)
     )
 
     switch (fetchContentDetail.apiResponse.status) {
       case 200:
-        const fileName = fetchContentDetail.body.filename
-        let folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+        folderDetail.fileName = fetchContentDetail.body.filename
+        folderDetail.folderParentsId = fetchContentDetail.body.parent_id ? [fetchContentDetail.body.parent_id] : []
+
+        let fetchLoopCondition = fetchContentDetail.body.parent_id !== null
```

Comment:
if you create a variable for a boolean test, the variable name should reflect what the test means, here, you can call it `hasReachRootWorkspace`. That way, it tells what `parent_id === null` means